frappe/handler.py

```diff
@@ -44,9 +44,13 @@ def handle():
 	cmd = frappe.local.form_dict.cmd
 	data = None
 
+	if not cmd:
+		raise frappe.DoesNotExistError
 	if cmd != "login":
-		data = execute_cmd(cmd)
-
+		try:
+			data = execute_cmd(cmd)
+		except Exception:
+			raise frappe.DoesNotExistError
```

Comment:
This isn't valid at all, it will wrap all exceptions happening in APIs entirely. 

Handle it better inside the function? if cmd doesn't resolve to a method then raise DoesNotExistError

Reply:
>  it will wrap all exceptions happening in APIs entirely.

Overlooked it 😅